src/sampling.jl

```diff
@@ -26,7 +26,7 @@ function Random.rand(bn::BayesNet, sampler::BayesNetSampler, nsamples::Integer)
     a = rand(bn, sampler)
     df = DataFrame()
     for cpd in bn.cpds
-        df[name(cpd)] = Array{typeof(a[name(cpd)])}(undef, nsamples)
+        df[!, name(cpd)] = Array{typeof(a[name(cpd)])}(undef, nsamples)
```

Comment:
Huh, I am not familiar with this use of `!`.  Neat

Reply:
@tawheeler Same here—as far as I can tell, the `!` syntax for in-place column assignment is specific to DataFrames. You can read up on their design and rules here: https://pkg.julialang.org/docs/DataFrames/AR9oZ/0.19.0/lib/indexing/#setindex!-1